src/renderer/components/BuyButton.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+
+import React, { useCallback } from "react";
+import { Trans } from "react-i18next";
+import Button from "~/renderer/components/Button";
+import { useHistory } from "react-router-dom";
+import { closeAllModal } from "~/renderer/actions/modals";
+import { useDispatch } from "react-redux";
+import type { TokenCurrency, CryptoCurrency } from "@ledgerhq/live-common/lib/types";
+import { getCryptoCurrencyById } from "@ledgerhq/live-common/lib/currencies";
+
+// NB Defaulting to ethereum since it's the only one throwing the NotEnoughGas currently
+const BuyButton = ({ currency: _c }: { currency?: CryptoCurrency | TokenCurrency }) => {
```

Comment:
there are I think two problem in this:

- the fact it's optional and infer it's "ethereum" is dangerous
- the fact it's `| TokenCurrency` does not seem covered, as you don't pass anything on other hand but just default to ethereum, it is not tested. 

TLDR: can we just `currency: CryptoCurrency` ?

- ethereum_classic could throw NotEnoughGas (as we converge to same impl), other new coin could in the future
- on the other side, you know what coin is involved as it's basically `mainAccount.currency`

Reply:
Indeed, it was originally optional because the first design had the cta inside the amount field, and the amount field didn't have the currency and I was lazy. With it being in the footer and having direct access to the currency, this is cleaner. Addressed.